src/pudl/transform/ferc1.py

```diff
@@ -2967,12 +2976,21 @@ def ferc1_transform_asset(
         Returns:
             transformed FERC Form 1 table.
         """
-        df = tfr_class(xbrl_metadata_json=xbrl_metadata_json[table_name]).transform(
+        if generic:
+            transformer = tfr_class(
+                xbrl_metadata_json=xbrl_metadata_json[table_name], table_id=table_id
```

Comment:
I don't think the generic transfer classes actually need the xbrl metadata. We could also remove the generic parameter and check if the `tfr_class[table_name]` is a `GenericPlantFerc1TableTransformer` instead.

Reply:
I didn't do this because I didn't want to deal with a circular import. We could put a deferred import at the top of this function, I don't really have a strong preference

Reply:
Ah good point. Lets stick with this 👍